### PR TITLE
Convert controller tests to RSpec

### PR DESCRIPTION
**Pull request by hannako**
Convert controller tests to RSpec as per alphagov/collections: Issue #2259
Trello: https://trello.com/c/DMSTbKug/1394-convert-model-tests-to-rspec

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/controllers/topics_controller_spec.rb [hunk 1]
```diff
@@ -12,15 +10,15 @@
...
+        expect(response.headers["Cache-Control"]).to eq("max-age=1800, public")
       end
     end
 
     it "returns a 404 status for GET topic with an invalid sector tag" do
       stub_content_store_does_not_have_item("/topic/oil-and-gas")
       get :show, params: { topic_slug: "oil-and-gas" }
 
-      assert_equal 404, response.status
+      expect(response.status).to eq(404)
```
Comment (sihugh) [inline spec/controllers/topics_controller_spec.rb:21]: I know this is a conversion (so only do it if you feel like it), but in the transition_landing_page_controller_spec we have this style (which I like):
`expect(response).to have_http_status(:success)` 
so we could switch this out to:
`expect(response).to have_http_status(:not_found)`

File: spec/controllers/taxons_controller_spec.rb [hunk 2]
```diff
@@ -1,9 +1,7 @@
-require "test_helper"
 require_relative "../../test/support/coronavirus_helper"
 
-describe TaxonsController do
+RSpec.describe TaxonsController do
   include SearchApiHelpers
-  include GovukAbTesting::MinitestHelpers
   include TaxonHelpers
 
   describe "GET show" do
```
Comment (sihugh) [inline spec/controllers/taxons_controller_spec.rb:7]: It doesn't seem to have converted this file (`assert_response` etc)

File: spec/controllers/step_nav_controller_spec.rb [hunk 3]
```diff
@@ -1,6 +1,4 @@
-require "test_helper"
-
-describe StepNavController do
+RSpec.describe StepNavController do
   it "returns a 404 when the page doesn't exist" do
     slug = SecureRandom.hex
     stub_content_store_does_not_have_item("/#{slug}")
```
Comment (sihugh) [inline spec/controllers/step_nav_controller_spec.rb:4]: looks like assert_response isn't picked up by the tool

assert_response 404 --> `expect(response).to have_http_status(:not_found)`

assert_response 403 --> `expect(response).to have_http_status(:forbidden)`

File: spec/controllers/subtopics_controller_spec.rb [hunk 4]
```diff
@@ -19,7 +19,7 @@
 
       get :show, params: { topic_slug: "oil-and-gas", subtopic_slug: "coal" }
 
-      assert_equal 404, response.status
+      expect(response.status).to eq(404)
```
Comment (sihugh) [inline spec/controllers/subtopics_controller_spec.rb:22]: `expect(response).to have_http_status(:not_found)` (if you like)

File: spec/controllers/second_level_browse_page_controller_spec.rb [hunk 5]
```diff
@@ -31,29 +29,22 @@
...
-          top_level_slug: "crime-and-justice",
-          second_level_slug: "frume",
-        },
-      )
-
+      params = {
+        top_level_slug: "crime-and-justice",
+        second_level_slug: "frume",
+      }
+      get :show, params: params
```
Comment (sihugh) [inline spec/controllers/second_level_browse_page_controller_spec.rb:47]: `expect(response).to have_http_status(:not_found)` 👇

File: spec/controllers/roles_controller_spec.rb [hunk 6]
```diff
@@ -19,7 +17,7 @@ def prime_minister
 
     it "when the content item exists" do
       get :show, params: { name: role }
-      assert_response :success
+      expect(response).to have_http_status(:success)
```
Comment (sihugh) [inline spec/controllers/roles_controller_spec.rb:20]: huh, it found `assert_response` here. Maybe it's looking for the :symbol invocation. 🤔

File: spec/controllers/roles_controller_spec.rb [hunk 7]
```diff
@@ -19,7 +17,7 @@ def prime_minister
...
       assert_template :show
     end
```
Comment (sihugh) [inline spec/controllers/roles_controller_spec.rb:23]: `expect(response).to have_http_status(:not_found)` 👇

**Review by sihugh — COMMENTED** [review]
Looks generally good, but there are a couple of `assert_response` calls that have been missed. If you have the inclination, it would be great to convert the `404` expectations to be `:not_found`s. I think it reads better, and would be consistent with the `:success` matches we have elsewhere here.

The other stuff looks great!

File: spec/controllers/taxons_controller_spec.rb (hunk 2 above)
Comment (hannako) [reply spec/controllers/taxons_controller_spec.rb:7]: hmm, I think that might be one of my classic speed rebase errors. will fix sorry!

**Comment by hannako** [comment]
Thanks @sihugh, I've fixed up those missing have_http_status matchers. Would you mind taking another look?

**Review by sihugh — APPROVED** [review]

**Pull request merged**